app/register.go

```diff
@@ -28,6 +28,12 @@ func (a *App) Register(w http.ResponseWriter, r *http.Request) {
 		},
 	}
 
+	// Compute secret hash based on client secret.
+	if a.AppClientSecret != "" {
```

Comment:
The a.AppClientSecret is not empty when we reach this if. Code inside this branch is not executed, you should just remove the if branch and the app will work perfectly.